lib/vmfloaty.rb

```diff
@@ -63,9 +64,13 @@ def run # rubocop:disable Metrics/AbcSize
           exit 1
         end
 
-        response = service.retrieve(verbose, os_types, use_token)
+        response = service.retrieve(verbose, os_types, use_token, options.ondemand)
+        request_id = response['request_id'] if options.ondemand
+        response = service.wait_for_request(verbose, request_id) if options.ondemand
+
         hosts = Utils.standardize_hostnames(response)
-        if options.json
+
+        if options.json || options.ondemand
```

Comment:
This doesn't make sense to me....generally the json option should print the same thing as the `format_hosts_output`, just in a pretty way for other tools to slurp up and use.

Reply:
I removed this and put it back to options.json. This was an artifact from me testing another capability that I did not add.